app/modules/user-management/api/auth-api.ts

```
@@ -0,0 +1,7 @@
+import { AxiosPromise } from 'axios';
+import { User } from './user-management-api';
+
+export interface AuthApi {
+  register(newUser: { user: User, password: string }): AxiosPromise<boolean>;
```

It would be better to throw an exception in case of unsuccessful registration, so `AxiosPromise<void>;` would be cleaner. Same for other parts.